citest/service_testing/cli_agent.py

```diff
@@ -156,7 +156,9 @@ def _args_to_full_commandline(self, args):
     This is an internal method used to ensure that the arguments are ultimately
     consistent if needed, regardless of how the agent is invoked.
     """
-    return [self.__program] + args
+    # The filter() here removes empty strings, which cause some cli
+    # commands to fail.
+    return filter(None, [self.__program] + args)
```

Comment:
There are valid `gcloud` commands where `name` is empty/null, such as `gcloud compute project-info describe`, so I tried to call it both with `None` and an empty string, neither of which worked. I pushed another change that removes this filter and fixes the `gcloud` command construction.